formulaic/models.py

```diff
@@ -36,6 +36,7 @@ def create_submission(self, cleaned_data, source=None, metadata=None):
             submission.date_created = timezone.now()
             submission.source = source
             submission.metadata = metadata or {}
+            submission.promo_source = submission.metadata.get('promo_source')
```

Comment:
We should take the time to pull this out of the `metadata` sooner.  Nothing in `metadata` should map to an actual field on the model.  Add another keyword argument to `create_submission` for this.